pkg/network/state.go

```diff
@@ -485,11 +500,36 @@ func (ns *networkState) storeClosedConnections(conns []ConnectionStats) {
 
 			if uint32(len(client.closedConnections)) >= ns.maxClosedConns {
 				stateTelemetry.closedConnDropped.Inc(c.Type.String())
+				// Drop if c is empty or closedConnections does not have empty connections
+				if isEmpty(c) || !isEmpty(client.closedConnections[len(client.closedConnections)-1]) {
+					continue
+				}
+				delete(client.closedConnectionsKeys, client.closedConnections[client.emptyConnsIndex].Cookie)
+				client.closedConnections[client.emptyConnsIndex] = c
+				client.closedConnectionsKeys[c.Cookie] = client.emptyConnsIndex
+				client.emptyConnsIndex++
+				continue
+			}
+
+			if isEmpty(c) {
+				client.closedConnections = append(client.closedConnections, c)
+				client.closedConnectionsKeys[c.Cookie] = len(client.closedConnections) - 1
+				continue
+			}
+
+			if client.emptyConnsIndex >= len(client.closedConnections) {
```

Comment:
Can `client.emptyConnsIndex > len(client.closedConnections)`?

Reply:
I thought it couldn't, but `TestSameKeyEdgeCases/TwoShortlivedConnectionsCrossing-1` fails if I don't add `>`.... I'll look into exactly what happens

Reply:
This really should be an invariant, `client.emptyConnsIndex <= len(client.closedConnections)`